### DISCUSSION TITLE
Broadcast Get Device ID using Master Read-Write over Serial

### DISCUSSION THREAD

**Discussion by hoangtn10** [ipmitool/ipmitool#311] [Q&A] [answered]
Hello,

I am looking for a way to send a Broadcast Get Device ID using the Master Read-Write command over Serial. If this is at all possible, any examples of the ipmitool command to do this, or the raw IPMI request would be greatly appreciated! 

Thank you!

**Comment by AlexanderAmelkin** [comment]
I don't quite understand what you need and/or what your BMC topology is.
`Master Write-Read` command is for sending raw I2C/SMBus commands to subordinate non-intelligent devices such as EEPROMs connected to a BMC.
`Get Device ID` is an IPMI command (i.e. not an I2C/SMBus command).
"Serial" is quite a third concept that has nothing to do with either I2C or IPMI (aside from Serial-over-LAN functionality).

Can you draw a picture describing what you're trying to access?

**Reply by hoangtn10** [reply]
Sorry for the confusion, I am still very new to the technology. 

For my setup, I can enter serial commands directly to the controller. For example, I can send `[18 00 01]` to retrieve the `Get Device ID` output. I can also bridge IPMI requests to the remote target address using `[18 00 34 ...]` (I think this is the send message command). I am trying to figure out how to use the `Broadcast Get Device ID`. The IPMI Specification said that the Broadcast 'Get Device ID' is not bridged but can be delivered to the IPMB using the Master Write-Read commands, but I am unsure what this means. 

If what I said does not make sense, please let me know. I am still trying to experiment/read up on this technology.

Thank you!

**Comment by doublechiang** [comment]
can you elaborate on "your setup"? And how do you use "ipmitool" to send the serial command?

**Reply by hoangtn10** [reply]
I am able to use `ipmitool –I serial-terminal –D /dev/ttyS0:9600 mc info` to get device id for example. This would send `[18 00 01]` over serial and parse the output.

**Reply by hoangtn10** [reply]
Reading over some more documentation, the ipmitool that I am using might be a special enhanced version. In which case it might not make sense to have this question posted here. If you guys are in agreement, I will close down this discussion thread.

**Reply by AlexanderAmelkin** [reply]
You may try to use an `i2c` subcommand with your `ipmitool` to construct a master write-read command.
The parameters to the `i2c` subcommand depends highly on your hardware setup and I can't tell you the values you need to use.

**Comment by doublechiang** [comment, marked as answer]
Please close the issue when it's being resolved.
